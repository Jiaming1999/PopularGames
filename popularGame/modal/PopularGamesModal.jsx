/* eslint-disable import/prefer-default-export */
import { BASE_URL } from '../env/env';

/**
 * fetching profile data from backend api
 * return interface used for other files
 *
 */
export const getPopularGame = async (limit = 100) => {
  const url = `${BASE_URL}popular?limit=${limit}`;

  try {
    const response = await fetch(
      url,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      },
    );
    const json = await response.json();
    return json;
  } catch (e) {
    throw new Error(e);
  }
};
