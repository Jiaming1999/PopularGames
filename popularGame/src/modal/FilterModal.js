/* eslint-disable import/prefer-default-export */
import { BASE_URL } from '../../env/env';

/**
 * parsing json file into object array stored
 * @param {*} json
 * @returns all games information
 */
export const parseGamesByAttr = (json) => {
  if (json === undefined) {
    throw new Error('fail to parse');
  }
  const allGames = json.map((game) => game);
  return allGames;
};

/**
 * Get game information by specific game genre and platform
 * @param {*} type
 * @param {*} category
 * @param {*} value
 * @returns
 */
export const getGameByAttr = async (type, category, value) => {
  const url = `${BASE_URL}${type}/rank?${category}=${value}`;

  try {
    const response = await fetch(
      url,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      },
    );
    const json = await response.json();
    return parseGamesByAttr(json);
  } catch (e) {
    throw new Error(e);
  }
};
