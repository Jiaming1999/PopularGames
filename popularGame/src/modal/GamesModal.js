/* eslint-disable import/prefer-default-export */
import { BASE_URL } from '../../env/env';

/**
 * parsing json file into object array stored
 * @param {*} json
 * @returns all games information
 */
export const parseGames = (json) => {
  if (json === undefined) {
    throw new Error('fail to parse');
  }
  const allGames = json.map((game) => game);
  return allGames;
};

/**
 * fetching profile data from backend api
 * return data used for other files/
 */
export const getGame = async (limit = 100, type) => {
  const url = `${BASE_URL}${type}?limit=${limit}`;

  try {
    const response = await fetch(
      url,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      },
    );
    const json = await response.json();
    return parseGames(json);
  } catch (e) {
    throw new Error(e);
  }
};
