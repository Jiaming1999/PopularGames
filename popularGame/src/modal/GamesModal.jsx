/* eslint-disable import/prefer-default-export */
import { BASE_URL } from '../../env/env';

/**
 * fetching profile data from backend api
 * return data used for other files/
 */
export const getPopularGame = async (limit = 100, type) => {
  const url = `${BASE_URL}${type}?limit=${limit}`;

  try {
    const response = await fetch(
      url,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      },
    );
    const json = await response.json();
    return json;
  } catch (e) {
    throw new Error(e);
  }
};
