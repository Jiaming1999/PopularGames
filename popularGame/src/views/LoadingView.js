/* eslint-disable react/jsx-filename-extension */
// eslint-disable-next-line no-use-before-define
import * as React from 'react';

import {
  Placeholder,
  PlaceholderMedia,
  PlaceholderLine,
  Fade,
} from 'rn-placeholder';

/**
 * Loading View for all data loading action
 * @returns
 */
const LoadingView = () => (
  <Placeholder
    Animation={Fade}
    Left={PlaceholderMedia}
    Right={PlaceholderMedia}
  >
    <PlaceholderLine width={80} />
    <PlaceholderLine />
    <PlaceholderLine width={30} />
  </Placeholder>
);

export default LoadingView;
