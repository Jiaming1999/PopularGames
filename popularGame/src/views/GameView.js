/* eslint-disable react/jsx-filename-extension */
import * as React from 'react';
import { useState } from 'react';
import {
  StyleSheet, ScrollView, TextInput, Button, TouchableOpacity, Image,
} from 'react-native';
import { PropTypes } from 'prop-types';
import {
  MaterialCommunityIcons, EvilIcons, MaterialIcons, AntDesign,
} from '@expo/vector-icons';
import LoadingView from './LoadingView';
import { POPULAR_MAX } from '../../env/env';
import { MAX_GAMES, MIN_GAMES } from '../constants/Numbers';
import { Text, View } from '../components/Themed.tsx';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  title: {
    paddingTop: 10,
    fontSize: 18,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  card: {
    width: '98%',
    flexShrink: 1,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#eee',
    margin: 5,
    padding: 5,
  },
  text: {
    fontSize: 15,
  },
  row: {
    flexDirection: 'row',
    padding: 5,
  },
  input: {
    width: '70%',
    margin: 15,
    height: 40,
    borderColor: '#eee',
    borderWidth: 1,
  },
  btn: {
    alignSelf: 'center',
    marginRight: 15,
  },
  image: {
    width: '90%',
    height: 200,
  },
});

/**
 * Card component for displaying game information
 * game: information for a single game from popular or top100 collection
 */
const GameInfo = (props) => {
  const { game, navigation } = props;

  const Score = () => {
    const items = [];
    const num = game.score === 'N/A' ? 0 : Math.ceil(game.score);
    for (let i = 0; i < num; i += 1) {
      items.push(<EvilIcons name="star" size={20} color="black" />);
    }
    return (
      <View style={styles.row}>
        {items}
      </View>
    );
  };

  return (
    <TouchableOpacity
      title="gamecard"
      style={styles.card}
      onPress={() => navigation.navigate('Article', game)}
    >
      <Image
        style={styles.image}
        source={{
          uri: game.thumbnail,
        }}
      />
      <View>
        <Text style={styles.title}>
          {
            game.title
          }
        </Text>
        <Text style={styles.text}>
          <Score />
        </Text>
        <Text style={styles.text}>
          <MaterialCommunityIcons name="typewriter" size={24} color="black" />
          {
            `Edited By ${game.editor}`
          }
        </Text>
        <Text style={styles.text}>
          <MaterialIcons name="category" size={24} color="black" />
          {
            `${game.genres}`
          }
        </Text>
        <Text style={styles.text}>
          <MaterialIcons name="gamepad" size={24} color="black" />
          {
            `On ${game.platforms}`
          }
        </Text>
        <Text style={styles.text}>
          <AntDesign name="calendar" size={24} color="black" />
          {
            game.release
          }
        </Text>
      </View>
    </TouchableOpacity>
  );
};

GameInfo.propTypes = {
  game: PropTypes.shape({
    title: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired,
    editor: PropTypes.string.isRequired,
    release: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
    developers: PropTypes.string.isRequired,
    genres: PropTypes.arrayOf(PropTypes.string),
    platforms: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

/**
 * display component for game list
 * @param {Filter box} Filter
 * @param {*} data
 * @param {*} navigation
 * @returns
 */
export const GameList = (props) => {
  const { data, navigation } = props;
  return (
    <ScrollView>
      {data.length !== 0 ? data.map((game) => (
        <GameInfo game={game} key={game.title} navigation={navigation} />
      )) : (
        <>
          <MaterialCommunityIcons style={styles.container} name="flask-empty-outline" size={24} color="black" />
          <Text style={styles.title}>No Data</Text>
        </>
      )}
    </ScrollView>
  );
};

GameList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
};

/**
 * Componet rendering a tab page of game information list
 * data: list of game information
 * setLimit: the number of game displayed
 */
const GameView = (props) => {
  const { data, setLimit, navigation } = props;
  const [value, setValue] = useState(POPULAR_MAX);

  if (!data) {
    return (
      <LoadingView />
    );
  }

  const Filter = () => (
    <>
      <View style={styles.row}>
        <TextInput
          style={styles.input}
          placeholder="limit"
          value={value}
          onChangeText={(text) => {
            setValue(text);
          }}
        />
        <View style={styles.btn}>
          <Button
            style={styles.btn}
            title="submit"
            color="#bf1313"
            onPress={() => {
              let tempValue;
              if (value > MAX_GAMES || value < MIN_GAMES || !value) {
                tempValue = MAX_GAMES;
              } else {
                tempValue = value;
              }
              setLimit(tempValue);
            }}
          />
        </View>
      </View>
    </>
  );

  return (
    <View style={styles.container}>
      <Filter />
      <GameList data={data} navigation={navigation} />
    </View>
  );
};

GameView.defaultProps = {
  data: undefined,
};

GameView.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
  setLimit: PropTypes.func.isRequired,
};

export default GameView;
