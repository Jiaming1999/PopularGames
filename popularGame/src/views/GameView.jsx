import * as React from 'react';
import { useState } from 'react';
import {
  StyleSheet, ScrollView, TextInput, Button,
} from 'react-native';
import { Avatar } from 'react-native-elements';
import { PropTypes } from 'prop-types';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import LoadingView from './LoadingView';
import { POPULAR_MAX } from '../../env/env';

import { Text, View } from '../components/Themed.tsx';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  title: {
    paddingTop: 10,
    fontSize: 12,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  card: {
    width: '98%',
    flexShrink: 1,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#eee',
    margin: 5,
    padding: 5,
  },
  text: {
    fontSize: 12,
  },
  row: {
    flexDirection: 'row',
  },
  input: {
    width: '70%',
    margin: 15,
    height: 40,
    borderColor: '#eee',
    borderWidth: 1,
  },
  btn: {
    alignSelf: 'center',
    marginRight: 15,
  },
});

/**
 * Card component for displaying game information
 * game: information for a single game from popular or top100 collection
 */
const GameInfo = (props) => {
  const { game, navigation } = props;
  return (
    <View style={styles.card}>
      <View style={styles.row}>
        <Avatar
          size="large"
          containerStyle={{ margin: 5, marginTop: 10 }}
          title={game.title}
          source={{ uri: game.thumbnail }}
          onPress={() => navigation.navigate('Article', game)}
        />
        <View>
          <Text style={styles.title}>
            {
              game.title
            }
          </Text>
          <Text style={styles.text}>
            {
              `Rated Score:${game.score}`
            }
          </Text>
          <Text style={styles.text}>
            {
              `By ${game.editor}`
            }
          </Text>
          <Text style={styles.text}>
            {
              `On ${game.platforms}`
            }
          </Text>
          <Text style={styles.text}>
            {
              `From ${game.developers}`
            }
          </Text>
          <Text style={styles.text}>
            {
              game.release
            }
          </Text>
        </View>
      </View>
    </View>
  );
};

GameInfo.propTypes = {
  game: PropTypes.shape({
    title: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired,
    editor: PropTypes.string.isRequired,
    release: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
    developers: PropTypes.arrayOf(PropTypes.string),
    platforms: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

/**
 * Componet rendering a tab page of game information list
 * data: list of game information
 * setLimit: the number of game displayed
 */
const GameView = (props) => {
  const { data, setLimit, navigation } = props;
  const [value, setValue] = useState(POPULAR_MAX);

  if (!data) {
    return (
      <LoadingView />
    );
  }

  const Filter = () => (
    <>
      <View style={styles.row}>
        <TextInput
          style={styles.input}
          placeholder="limit"
          value={value}
          onChangeText={(text) => {
            setValue(text);
          }}
        />
        <View style={styles.btn}>
          <Button
            style={styles.btn}
            title="submit"
            color="#bf1313"
            onPress={() => {
              let tempValue;
              if (value > 100 || value < 0 || !value) {
                tempValue = 100;
              } else {
                tempValue = value;
              }
              setLimit(tempValue);
            }}
          />
        </View>
      </View>
    </>
  );

  return (
    <View style={styles.container}>
      <Filter />
      <ScrollView>
        {data.length !== 0 ? data.map((game) => (
          <GameInfo game={game} key={game.title} navigation={navigation} />
        )) : (
          <>
            <MaterialCommunityIcons style={styles.container} name="flask-empty-outline" size={24} color="black" />
            <Text style={styles.title}>No Data</Text>
          </>
        )}
      </ScrollView>
    </View>
  );
};

GameView.defaultProps = {
  data: undefined,
};

GameView.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
  setLimit: PropTypes.func.isRequired,
};

export default GameView;
