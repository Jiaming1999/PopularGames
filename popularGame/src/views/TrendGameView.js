/* eslint-disable react/jsx-filename-extension */
import * as React from 'react';
import {
  StyleSheet, ScrollView, Button,
} from 'react-native';
import { PropTypes } from 'prop-types';
import { View, Text } from '../components/Themed.tsx';
import { GameInfo } from './GameView';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  title: {
    paddingTop: 10,
    fontSize: 18,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  card: {
    width: '98%',
    flexShrink: 1,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: '#eee',
    margin: 5,
    padding: 5,
  },
  text: {
    fontSize: 15,
  },
  row: {
    flexDirection: 'row',
    padding: 5,
  },
  input: {
    width: '70%',
    margin: 15,
    height: 40,
    borderColor: '#eee',
    borderWidth: 1,
  },
  btn: {
    alignSelf: 'center',
    marginRight: 15,
  },
  image: {
    width: '90%',
    height: 200,
  },
});

/**
 * Game view display for a trend game review page
 * Support check games/genres/platform for most like and least like
 * @param {function} setType set the type of which filtering used
 * @param {object} data response from filtering
 * @param {array of games} gameData games satisfied the filtering
 * @returns
 */
const TrendGameView = (props) => {
  const {
    setType, data, gameData, navigation,
  } = props;

  const handlePressGenreMost = () => {
    setType('mostgenre');
  };

  const handlePressPlatfromMost = () => {
    setType('mostplatform');
  };

  const handlePressGenreLeast = () => {
    setType('leastgenre');
  };

  const handlePressPlatfromLeast = () => {
    setType('leastplatform');
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>
        {data ? data.response : 'Click to see result'}
      </Text>
      <View style={styles.row}>
        <Button
          style={styles.btn}
          title="MostGenre"
          color="#bf1313"
          onPress={handlePressGenreMost}
        />
        <Button
          style={styles.btn}
          title="MostPlatform"
          color="#bf1313"
          onPress={handlePressPlatfromMost}
        />
      </View>
      <View style={styles.row}>
        <Button
          style={styles.btn}
          title="LeastGenre"
          color="#bf1313"
          onPress={handlePressGenreLeast}
        />
        <Button
          style={styles.btn}
          title="LeastPlatform"
          color="#bf1313"
          onPress={handlePressPlatfromLeast}
        />
      </View>
      <ScrollView>
        {gameData ? gameData.map((game) => (
          <GameInfo game={game} key={game.title} navigation={navigation} />
        )) : <Text>See Games Satisfied Those Trend</Text>}
      </ScrollView>
    </View>
  );
};

TrendGameView.propTypes = {
  setType: PropTypes.func.isRequired,
  data: PropTypes.shape({
    response: PropTypes.string.isRequired,
  }).isRequired,
  gameData: PropTypes.shape({
    map: PropTypes.func.isRequired,
  }).isRequired,
};

export default TrendGameView;
