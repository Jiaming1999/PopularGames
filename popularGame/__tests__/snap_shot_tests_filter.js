/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
import * as React from 'react';
import renderer from 'react-test-renderer';
import api from '../src/util/api';
import {
  MOST_GENRE_DATA, MOST_PLATFORM_DATA, LEAST_GENRE_DATA, LEAST_PLATFORM_DATA,
} from '../mockData';
import { parseGamesByAttr } from '../src/modal/FilterModal';
import { Result } from '../src/views/TrendGameView';

jest.useFakeTimers();

beforeEach(() => {
  fetch.resetMocks();
});

/**
 * snap shot test for Normally display most genre result for  games
 */
test('returns result for mock popular game', () => {
  fetch.mockResponseOnce(JSON.stringify([MOST_GENRE_DATA]));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return api('/filter/popular?type=mostgenre')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();
      const data = parseGamesByAttr([onResponse.mock.calls[0][0][0]]);
      const tree = renderer.create(<Result data={data[0]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

/**
 * snap shot test for Normally display most platform result for popular games
 */
test('returns result for mock popular game', () => {
  fetch.mockResponseOnce(JSON.stringify([MOST_PLATFORM_DATA]));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return api('/filter/popular?type=mostplatform')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();
      const data = parseGamesByAttr([onResponse.mock.calls[0][0][0]]);
      const tree = renderer.create(<Result data={data[0]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

/**
 * snap shot test for Normally display least genre result for top100 games
 */
test('returns result for mock popular game', () => {
  fetch.mockResponseOnce(JSON.stringify([LEAST_GENRE_DATA]));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return api('/filter/top100?type=mostgenre')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();
      const data = parseGamesByAttr([onResponse.mock.calls[0][0][0]]);
      const tree = renderer.create(<Result data={data[0]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

/**
 * snap shot test for Normally display least platform result for top100 games
 */
test('returns result for mock popular game', () => {
  fetch.mockResponseOnce(JSON.stringify([LEAST_PLATFORM_DATA]));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return api('/filter/top100?type=mostplatform')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();
      const data = parseGamesByAttr([onResponse.mock.calls[0][0][0]]);
      const tree = renderer.create(<Result data={data[0]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
