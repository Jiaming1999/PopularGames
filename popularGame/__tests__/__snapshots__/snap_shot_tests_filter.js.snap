// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns result for mock popular game 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flexDirection": "row",
        "padding": 5,
      },
    ]
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
        },
      ]
    }
  >
    FPS
  </Text>
  <Text />
</View>
`;

exports[`returns result for mock popular game 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flexDirection": "row",
        "padding": 5,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
        },
      ]
    }
  >
    PlayStation4
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`returns result for mock popular game 3`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flexDirection": "row",
        "padding": 5,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
        },
      ]
    }
  >
    RPG
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`returns result for mock popular game 4`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "flexDirection": "row",
        "padding": 5,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
        },
      ]
    }
  >
    XBOX360
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
